crates/client/data-availability/src/avail/mod.rs

```diff
@@ -50,41 +89,47 @@ impl DaClient for AvailClient {
 }
 
 impl AvailClient {
-    async fn publish_data(&self, bytes: &BoundedVec<u8>) -> Result<H256> {
+    async fn publish_data(&self, bytes: &BoundedVec<u8>) -> Result<()> {
+        let mut ws_client = self.ws_client.lock().await;
```

Comment:
why do we need a lock ? afaik there is no concurrency possible

Reply:
There is no concurrency possible, but we need to update the client with the new instance during which no one should access it. If its inside a method with non mutable reference, I wanted to keep it behind mutex guard so it is updated safely. Let me know if you think it should be implemented differently.

Reply:
Fair enough, thanks for the explanation